app/models/foreman_virt_who_configure/output_generator.rb

```diff
@@ -42,19 +79,29 @@ def virt_who_output
 rhsm_encrypted_password=$user_password
 rhsm_prefix=/rhsm
 EOF
+  if [ $? -ne 0 ]; then result_code=$(($result_code|#{error_code(:virt_who_config_file_issue)})); fi
 
-echo "Creating sysconfig virt-who configuration.."
-cat > /etc/sysconfig/virt-who << EOF
+  echo "Creating sysconfig virt-who configuration.."
+  cat > /etc/sysconfig/virt-who << EOF
 VIRTWHO_SATELLITE6=1
 VIRTWHO_DEBUG=#{config.debug? ? 1 : 0}
 VIRTWHO_INTERVAL=#{config.interval * 60}#{proxy_strings}
 EOF
+  if [ $? -ne 0 ]; then result_code=$(($result_code|#{error_code(:virt_who_sysconfig_file_issue)})); fi
+
+  echo "Enabling and restarting the virt-who service"
+  chkconfig virt-who on || $(($result_code|#{error_code(:virt_who_chkconfig_issue)}))
+  service virt-who restart || $(($result_code|#{error_code(:virt_who_service_issue)}))
+else
+  result_code=$(($result_code|#{error_code(:virt_who_too_old)}))
+fi
 
-echo "Enabling and restarting the virt-who service"
-chkconfig virt-who on
-service virt-who restart
+if [ $result_code -ne 0 ]; then
+  echo "Errors during configuration, return code is $result_code"
```

Comment:
The code can be used to find out what errors happened. The error code 3 represents there were two errors, with code 1 and 2. I'd like to avoid having bash function in a script that would find out what strings to print. The error code will be much more useful when we use REX for deployment where we could reconstruct error messages based on it. If you want to reuse that in hammer plugin somehow, I can add such hash into API. Do you have some suggestion on format/structure?